backend/clickhouse/logs.go

```diff
@@ -72,29 +72,65 @@ func (client *Client) BatchWriteLogRows(ctx context.Context, logRows []*LogRow)
 	return batch.Send()
 }
 
-const Limit int = 100
+const LogsLimit int = 100
 const KeyValuesLimit int = 50
 
-func (client *Client) ReadLogs(ctx context.Context, projectID int, params modelInputs.LogsParamsInput, after *string) (*modelInputs.LogsPayload, error) {
-	sb, err := makeSelectBuilder("Timestamp, UUID, SeverityText, Body, LogAttributes, TraceId, SpanId, SecureSessionId", projectID, params, after)
-	if err != nil {
-		return nil, err
-	}
+const OrderBackward = "Timestamp ASC, UUID ASC"
+const OrderForward = "Timestamp DESC, UUID DESC"
 
-	sb.OrderBy("Timestamp DESC, UUID DESC").Limit(Limit + 1)
+type Pagination struct {
+	After          *string
+	AfterOrEqualTo *string // currently only used internally
```

Comment:
looks like this overlap with the `before` results (the row at the cursor is included in both `afterOrEqual` and `before` results. guessing that there is some other logic that removes the duplicate row, but just wanted to double check.

Reply:
Good catch. I think in my testing that apollo was handling deduplication. 
I think the union below should be change from `UNION ALL` to `UNION DISTINCT`

I'll add a test to confirm.

Reply:
Added 8ee38e023cec42da092964b113f5ec49c9d737ee

It removes the `afterOrEqual` flow and added an equal sub select to the union. It's effectively:

SELECT (Before Cursor) UNION ALL SELECT (Equal to cursor) UNION ALL SELECT (After cursor)

This removes the need to use distinct.